projects/packages/publicize/src/jetpack-social-settings/class-settings.php

```diff
@@ -0,0 +1,263 @@
+<?php
+/**
+ * Settings class.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize\Jetpack_Social_Settings;
+
+use Automattic\Jetpack\Modules;
+use Automattic\Jetpack\Publicize\Social_Image_Generator\Templates;
+
+/**
+ * This class is used to get and update Jetpack_Social_Settings.
+ * Currently supported features:
+ *      - Social Image Generator
+ *      - Auto Conversion
+ */
+class Settings {
+	/**
+	 * Name of the database option.
+	 *
+	 * @var string
+	 */
+	const OPTION_PREFIX = 'jetpack_social_';
+	// cSpell:ignore AUTOCONVERT
+	const AUTOCONVERT_IMAGES       = 'autoconvert_images';
+	const IMAGE_GENERATOR_SETTINGS = 'image_generator_settings';
+
+	const DEFAULT_IMAGE_GENERATOR_SETTINGS = array(
+		'enabled'  => false,
+		'template' => Templates::DEFAULT_TEMPLATE,
+	);
+
+	const DEFAULT_AUTOCONVERT_IMAGES_SETTINGS = array(
+		'enabled' => true,
+	);
+
+	/**
+	 * Migrate old options to the new settings. Previously SIG settings were stored in the
+	 * jetpack_social_image_generator_settings option. Now they are stored in the jetpack_social_settings
+	 * together with the auto conversion settings.
+	 *
+	 * TODO: Work out if this is possible on plugin upgrade
+	 *
+	 * @return void
+	 */
+	private function migrate_old_option() {
+		$auto_conversion_settings = get_option( 'jetpack_social_settings' );
+		if ( ! empty( $auto_conversion_settings ) ) {
+			update_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES, array( 'enabled' => ! empty( $auto_conversion_settings['image'] ) ) );
+			delete_option( 'jetpack_social_settings' );
+		}
+
+		$sig_settings = get_option( 'jetpack_social_image_generator_settings' );
+		$enabled      = false;
+		$template     = Templates::DEFAULT_TEMPLATE;
+
+		if ( isset( $sig_settings['defaults']['template'] ) ) {
+			$template = $sig_settings['defaults']['template'];
+		}
+
+		if ( isset( $sig_settings['enabled'] ) ) {
+			$enabled = $sig_settings['enabled'];
+		}
+
+		if ( ! isset( $sig_settings['template'] ) ) {
+			update_option(
+				self::OPTION_PREFIX . self::IMAGE_GENERATOR_SETTINGS,
+				array(
+					'enabled'  => $enabled,
+					'template' => $template,
+				)
+			);
+		}
+	}
+
+	/**
+	 * Register the settings.
+	 *
+	 * @return void
+	 */
+	public function register_settings() {
+		register_setting(
+			'general',
+			self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES,
+			array(
+				'default'      => array(
+					'enabled' => true,
+				),
+				'show_in_rest' => array(
+					'schema' => array(
+						'type'       => 'object',
+						'properties' => array(
+							'enabled' => array(
+								'type' => 'boolean',
+							),
+						),
+					),
+				),
+				'type'         => 'boolean',
```

Comment:
I think something might be wrong here, this type should be `object`. Do you think that's how it ended up being the empty string? I think maybe something has been cast as a boolean and it's evaluated to false.